maps-compose/src/main/java/com/google/maps/android/compose/GoogleMap.kt

```diff
@@ -123,31 +130,166 @@ public fun GoogleMap(
+                mapProperties = currentMapProperties,
+                mapUiSettings = currentUiSettings,
+            )
+
+            MapClickListenerUpdater()
+
+            CompositionLocalProvider(
+                LocalCameraPositionState provides currentCameraPositionState,
+            ) {
+                currentContent?.invoke()
+            }
+        }
+
+        currentComposition.apply {
+            if(mapReused) {
+                setContentWithReuse(mapCompositionContent)
+            } else {
+                setContent(mapCompositionContent)
+            }
+        }
+
+        // Set this after composition is started.
+        mapClickListeners = clickListeners
+    }
+
+    AndroidView(
+        modifier = modifier,
+        factory = {
+            Log.d(TAG, "Factory")
+            debugMapReused = false
+            MapView(context, googleMapOptionsFactory()).also { mapView ->
+                mapLifecycleController = MapViewLifecycleController(
+                    isMapReused = false,
+                    mapView = mapView
+                )
+                componentCallbacks = mapView.componentCallbacks()
+            }
+        },
+        onReset = { mapView ->
+            mapView.log("onReset")
+            // Deactivate composition to save resources
+            context.unregisterComponentCallbacks(componentCallbacks)
+            mapLifecycleController!!.onLifecycleDetached()
+            composition?.deactivate()
+            // Call onStop/onPause or something? Set map type to None to save resources? Because the MapView is detached.
+        },
+        onRelease = { mapView ->
+            mapView.log("onRelease")
+            context.unregisterComponentCallbacks(componentCallbacks)
+            // Dispose composition
+            composition?.dispose()
+            // Invoke onDestroy + remove lifecycle callbacks for the MapView
+            mapLifecycleController!!.onDestroy()
+            // Clean up MapView.
+            mapView.removeAllViews()
+        },
+        update = { mapView ->
+            mapView.log("update")
+            if (mapLifecycleController == null) {
+                debugMapReused = true
+                mapLifecycleController = MapViewLifecycleController(
+                    isMapReused = true,
+                    mapView = mapView
                 )
+            }
 
-                MapClickListenerUpdater()
+            if (componentCallbacks == null) {
+                componentCallbacks = mapView.componentCallbacks()
+            }
 
-                CompositionLocalProvider(
-                    LocalCameraPositionState provides currentCameraPositionState,
-                ) {
-                    currentContent?.invoke()
+            mapLifecycleController!!.lifecycle = lifecycle
+
+            // Create Composition
+            if(!isCompositionSet) {
+                isCompositionSet = true
+                mapUpdaterScope.launch {
```

Comment:
Are you open to making this launch UNDISPATCHED as an optimization? I think the only suspending thing here is `awaitMap()` and we want that to happen ASAP, so barrel through until we're there.

In my experience even Dispatchers.Main.immediate seems to use some kind of dispatch or yield() where it may suspend during launch.

Reply:
For sure

Reply:
Thank you for the explanation.